Editor/Overlays/CinemachineToolbarOverlay.cs

```diff
@@ -1,138 +1,169 @@
+                    + "/Editor/EditorResources/TrackedObjectOffset.png"),
+                tooltip = "Tracked Object Offset Tool",
+            };
         }
     }
-    
+
+#if UNITY_2022_1_OR_NEWER
     /// <summary>
-    /// Creates a toggle tool on the Cinemachine toolbar.
+    /// To add your custom tools (EditorToolbarElement) to the Cinemachine Tool Settings toolbar,
+    /// set CinemachineToolSettingsOverlay.customToolbarItems with your custom tools' IDs.
+    ///
+    /// By default, CinemachineToolSettingsOverlay.customToolbarItems is null.
     /// </summary>
-    abstract class CinemachineEditorToolbarToggle : EditorToolbarToggle
+    [Overlay(typeof(SceneView), "Cinemachine Tool Settings")]
+    [Icon("Packages/com.unity.cinemachine/Gizmos/cm_logo.png")]
+    public class CinemachineToolSettingsOverlay : Overlay, ICreateToolbar
     {
-        protected CinemachineEditorToolbarToggle()
+        static readonly string[] k_CmToolbarItems = { FreelookRigSelection.id };
+
+        public override VisualElement CreatePanelContent() => CreateContent(Layout.HorizontalToolbar);
+
+        /// <summary>Set this with your custom tools' IDs.</summary>
+        public static string[] customToolbarItems = null;
+        public IEnumerable<string> toolbarElements
         {
-            CinemachineSceneToolUtility.RegisterToolHandlers(GetType(), isOn => value = isOn, 
-                display => style.display = display ? DisplayStyle.Flex : DisplayStyle.None);
+            get
+            {
+                if (customToolbarItems != null)
+                {
+                    var toolbarItems = new List<string>(k_CmToolbarItems);
+                    toolbarItems.AddRange(customToolbarItems);
+                    return toolbarItems;
+                }
+                
+                return k_CmToolbarItems;
+            }
         }
     }
-    
-    [EditorToolbarElement(id, typeof(SceneView))]
-    class SoloVcamTool : CinemachineEditorToolbarToggle
+#else
+    // Not extendable overlay
+    [Overlay(typeof(SceneView), "Cinemachine Tool Settings")]
+    [Icon("Packages/com.unity.cinemachine/Gizmos/cm_logo.png")]
+    class CinemachineToolSettingsOverlay : ToolbarOverlay
     {
-        public const string id = "SoloVcamTool/Toggle";
-
-        public SoloVcamTool()
-        {
-            this.RegisterValueChangedCallback(
-                v => CinemachineSceneToolUtility.SetSolo(v.newValue));
-            onIcon = EditorGUIUtility.IconContent("animationvisibilitytoggleon@2x").image as Texture2D;
-            offIcon = EditorGUIUtility.IconContent("animationvisibilitytoggleoff@2x").image as Texture2D;
-            tooltip = "Solo Vcam Tool";
-        }
+        public CinemachineToolSettingsOverlay() : base(FreelookRigSelection.id) {}
     }
-    
+#endif
     
     [EditorToolbarElement(id, typeof(SceneView))]
     class FreelookRigSelection : EditorToolbarDropdown
```

Comment:
Rig selection tool looks ridiculous.  Should we remove it until the bug is fixed and it can look good?  
Other option: can we make icons instead of text? Will they show?

![image](https://user-images.githubusercontent.com/6424658/143087183-50fadca8-3d88-4d64-8a1e-853019be022e.png)

Reply:
I agree. I put the whole CinemachineSceneSettingsOverlay and FreelookRigSelection under UNITY_2022_1_OR_NEWER.